addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,339 @@
+
+    @api.depends('order_ids')
+    def _compute_invoice_amounts(self):
+        for wizard in self:
+            # _origin -> See https://github.com/odoo/odoo/pull/133836
+            wizard.amount_invoiced = sum(wizard.order_ids._origin.mapped('amount_invoiced'))
+            wizard.amount_to_invoice = sum(wizard.order_ids._origin.mapped('amount_to_invoice'))
+
+    # === ONCHANGE METHODS ===#
+
+    @api.onchange('advance_payment_method')
+    def _onchange_advance_payment_method(self):
+        if self.advance_payment_method == 'percentage':
+            amount = self.default_get(['amount']).get('amount')
+            return {'value': {'amount': amount}}
+
+    # === CONSTRAINT METHODS ===#
+
+    def _check_amount_is_positive(self):
+        for wizard in self:
+            if (wizard.advance_payment_method == 'percentage' and float_compare(wizard.amount, 0, precision_rounding=wizard.currency_id.rounding) <= 0) \
+                    or (wizard.advance_payment_method == 'fixed' and float_compare(wizard.fixed_amount, 0, precision_rounding=wizard.currency_id.rounding) <= 0):
+                raise UserError(_('The value of the down payment amount must be positive.'))
+
+    # === ACTION METHODS ===#
+
+    def create_invoices(self):
+        self._check_amount_is_positive()
+        invoices = self._create_invoices(self.order_ids)
+        return self.order_ids.action_view_invoice(invoices=invoices)
+
+    def view_draft_invoices(self):
+        return {
+            'name': _('Draft Invoices'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'tree',
+            'views': [(False, 'list'), (False, 'form')],
+            'res_model': 'account.move',
+            'domain': [('state', '=', 'draft')],  # To override
+        }
+
+    def _create_invoices(self, orders):
+        if self.advance_payment_method == 'delivered':
+            return orders._create_invoices(final=True, grouped=self._needs_to_group_on_invoice())
+
+        return self.create_down_payment_invoice()
+
+    def create_down_payment_invoice(self):
+        self.order_ids.ensure_one()
+        self = self.with_company(self.company_id)
+        order = self.order_ids
+
+        # Create down payment section if necessary
+        if not any(line.display_type and line.is_downpayment for line in order.order_line):
+            self.order_ids.with_context(no_log_for_new_lines=True)._create_new_order_line(
+                self._prepare_down_payment_section_values(order)
+            )
+
+        values, accounts = self._prepare_down_payment_lines_values(order)
+        down_payment_lines = self.order_ids.with_context(no_log_for_new_lines=True)._create_new_order_line(values)
+
+        invoice = self.env['account.move'].sudo().create(
+            self._prepare_invoice_values(order, down_payment_lines, accounts)
+        )
+
+        # Ensure the invoice total is exactly the expected fixed amount.
+        if self.advance_payment_method == 'fixed':
```

Comment:
No need for percentage ?

Reply:
In the percentage case this issue with the rounding and the tax does not occur. It is only to make sure that in the fixed case it does not end up being 250.01 when putting 250.